### ISSUE TITLE
Upload deb file to PPA

### ISSUE THREAD

**Issue by willprice** [GetPoplog/Seed#49]
Upload released deb files to https://launchpad.net/~getpoplog/+archive/ubuntu/poplog

**Comment by willprice** [comment]
Relevant material:
- https://help.launchpad.net/Packaging/PPA/Uploading
- https://packaging.ubuntu.com/html/packaging-new-software.html
- https://honk.sigxcpu.org/projects/git-buildpackage/manual-html/gbp.html

**Comment by willprice** [comment]
You cannot just upload a deb package to PPA. Instead you must provide a build recipe and some metadata in a `debian` folder which is used by build machines on launchpad to build the package. It is unclear to me whether the use of `bzr` is mandatory or not. A command, `dput`, is used to upload the `.dsc`, `.changes`, `.debian.tar.gz` files to launchpad.

**Comment by willprice** [comment]
Andrei suggested taking a look at OBS (https://openbuildservice.org/) as an alternative. This is a build service to package for the major platforms in a way that makes the resulting packages (relatively) easy to install.

**Comment by sfkleach** [comment]
> Andrei suggested taking a look at OBS (https://openbuildservice.org/) as an alternative. This is a build service to package for the major platforms in a way that makes the resulting packages (relatively) easy to install.

I note the slightly worrying restriction: "The OBS builds your packages in a jailed environment that the OBS setups from scratch each time and once your build succeeds it also rebuilds all the depending packages for you." We need to figure out what the "jail" is because of the current issues with Poplog and containers (due to memory management).

**Comment by willprice** [comment]
launchpad also does this, I suspect they are chroots.

**Comment by willprice** [comment]
Alternatively we can look at making our own deb/rpm repos that users can add. That seems simpler than getting something on launchpad.

**Comment by sfkleach** [comment]
Chroots should be fine - although it would be a good idea to check at some point. Firejail I haven't checked and definitely mean to.

**Comment by willprice** [comment]
I suppose we should figure this out before investing the effort to try and get running on one of these services. The last thing I want is to put the (large) amount of effort required to start building and distributing on one of these services and find out that we can't even compile poplog on it.

**Comment by sfkleach** [comment]
If we find it's a blocker, simply postpone it until we solve the memory management issue. 

And I found this in the docs: The default build environment for local builds is usually chroot. While this is simplest environment and is therefore easy and fast to handle, it has also a number of shortcomings. For one it is not safe against attacks, so you must not build sources or using build dependencies from a resource which you do not trust. Furthermore the environment is not fully isolated and runs not the kernel the target distribution runs. This means esp image builds and kernel/hardware specific builds may fail or won't produce the same result. The server side is usually set to inside of KVM therefore to avoid this.

**Comment by sfkleach** [comment]
So it is KVM and that means there's a real risk Poplog might not work. So it's definitely something to check before investing any time.

**Comment by willprice** [comment]
As part of writing the debian package, I ran the build under `dpkg-depcheck` to see what packages it uses, here's what it reported:

```
gcc-9
libxau6:amd64
libx11-6:amd64
libattr1:amd64
binutils-x86-64-linux-gnu
findutils
libgcc-9-dev:amd64
cpp-9
libbinutils:amd64
libx11-dev:amd64
coreutils
libxdmcp6:amd64
libgmp10:amd64
tzdata
libpcre2-8-0:amd64
libctf0:amd64
x11proto-dev
libc6-dev:amd64
libbsd0:amd64
libc-bin
make
libisl22:amd64
libxt6:amd64
libmpc3:amd64
libfakeroot:amd64
libxt-dev:amd64
linux-libc-dev:amd64
libc6:amd64
libacl1:amd64
libice6:amd64
libice-dev:amd64
libsm-dev:amd64
libmpfr6:amd64
libxcb1:amd64
libsm6:amd64
```

**Comment by willprice** [comment]
Addressed in #69

**Comment by willprice** [comment]
Builds are available at https://build.opensuse.org/package/show/home:willprice/poplog
although they're managed manually atm.

**Comment by willprice** [comment]
Need to figure out how to split `packaging/deb/control` into `poplog.dsc` for consumption on OBS. That file defines the source build.

**Comment by willprice** [comment]
Addressed in #72
